clio-server/src/main/scala/org/broadinstitute/clio/server/webservice/RejectionDirectives.scala

```diff
@@ -13,19 +13,30 @@ trait RejectionDirectives { self: JsonWebService =>
   }
 
   private val mapRejectionsToJsonHandler: RejectionHandler = {
-    RejectionHandler.default.mapRejectionResponse {
-      case httpResponse @ HttpResponse(
-            _,
-            _,
-            httpEntity: HttpEntity.Strict,
-            _
-          ) =>
-        // since all Akka default rejection responses are Strict this will handle all rejections
-        val message = RejectionResult(httpEntity.data.utf8String)
-        val jsonMessage = message.asJson.noSpaces
-        httpResponse.withEntity(ContentTypes.`application/json`, jsonMessage)
-      case httpResponse =>
-        httpResponse // pass through all other types of responses
-    }
+    RejectionHandler
+      .newBuilder()
+      .handle {
+        case ApiNotReadyRejection => {
+          complete(
+            StatusCodes.ServiceUnavailable -> "Clio is rebuilding search index, try again later"
```

Comment:
Would it be helpful/easy to include the time clio has been indexing here? It would be even more helpful to include the "progress" data, such as the date of the last processed record.

Reply:
Time since recovery started should be easy; time since last upsert less so (but I'll look into it)